diff --git a/src/watchers/FallbackWatcher.js b/src/watchers/FallbackWatcher.js
index e431e90cba1d0700a45b4b5e21c95c8fd1bb0eba..63b90417bf0c71d6ee013ddc36453eeb8f0b3ff9 100644
--- a/src/watchers/FallbackWatcher.js
+++ b/src/watchers/FallbackWatcher.js
@@ -116,6 +116,9 @@ class FallbackWatcher extends _AbstractWatcher.AbstractWatcher {
     if (this.watched[dir]) {
       return false;
     }
+    if (/(^|[\\/])(\.git|node_modules)([\\/]|$)/.test(dir)) {
+      return false;
+    }
     const watcher = _fs.default.watch(
       dir,
       {
